pkg/antctl/raw/test/command.go

```diff
@@ -0,0 +1,493 @@
+				k.Log("curl connectivity check command failed: %s", err)
+				success = false
+			}
+
+			if success {
+				k.Log("client pod %s was able to communicate with echo pod %s (%s)", clientPod.Name, echoName, echoIP)
+			} else {
+				k.Log("client pod %s was not able to communicate with echo pod %s (%s)", clientPod.Name, echoName, echoIP)
+			}
+
+			k.Relax()
+		}
+	}
+}
+
+func (k *k8sConnectivityParams) validatePodInternetConnectivity(ctx context.Context) error {
+	for _, clientPod := range k.clientPods.Items {
+		var (
+			srcPod  = k.namespace + "/" + clientPod.Name
+			success = true
+		)
+
+		k.Header("Validating connectivity from pod %s to the world (google.com)...", srcPod)
+		_, err := k.client.ExecInPod(ctx, k.namespace, clientPod.Name, clientDeploymentName, agnhostConnectCommand("google.com:80"))
+		if err != nil {
+			k.Log("Connectivity test from pod %s to google.com failed: %s", srcPod, err)
+			success = false
+		}
+
+		if success {
+			k.Log("Pod %s was able to connect to google.com", srcPod)
+		}
+
+		k.Relax()
+	}
+	return nil
+}
+
+func (k *k8sConnectivityParams) Relax() {
+	time.Sleep(2 * time.Second)
+}
+
+const podReadyTimeout = 5 * time.Minute
+
+func (k *k8sConnectivityParams) deleteDeployments(ctx context.Context, client k8sClientOperations) error {
+	k.Log("[%s] Deleting connectivity check deployments...", client.ClusterName())
+	client.DeleteNamespace(ctx, k.namespace, metav1.DeleteOptions{})
+
+	_, err := client.GetNamespace(ctx, k.namespace, metav1.GetOptions{})
+	if err == nil {
+		k.Log("[%s] Waiting for namespace %s to disappear", client.ClusterName(), k.namespace)
+		for err == nil {
+			time.Sleep(time.Second)
+			_, err = client.GetNamespace(ctx, k.namespace, metav1.GetOptions{})
+		}
+	}
+
+	return nil
+}
+
+func (k *k8sConnectivityParams) deploymentList() (srcList []string, dstList []string) {
+	srcList = []string{clientDeploymentName, echoSameNodeDeploymentName}
+	dstList = append(dstList, echoOtherNodeDeploymentName)
+
+	return srcList, dstList
+}
+
+type deploymentClients struct {
+	source      k8sClientOperations
+	destination k8sClientOperations
+}
+
+func (d *deploymentClients) clients() []k8sClientOperations {
+	return []k8sClientOperations{d.source}
+}
+
+func (k *k8sConnectivityParams) initClients(ctx context.Context) (*deploymentClients, error) {
+	c := &deploymentClients{
+		source:      k.client,
+		destination: k.client,
+	}
+
+	_, err := k.client.GetDaemonSet(ctx, k.antreaNamespace, agentDaemonSetName, metav1.GetOptions{})
+	if err != nil {
+		k.Log("Unable to determine status of Antrea DaemonSet.")
+		return nil, fmt.Errorf("Unable to determine status of antrea DaemonSet: %w", err)
+	}
+
+	return c, nil
+}
+
+func (k *k8sConnectivityParams) deploy(ctx context.Context) error {
+	var srcDeploymentNeeded, dstDeploymentNeeded bool
+
+	_, err := k.clients.source.GetNamespace(ctx, k.namespace, metav1.GetOptions{})
+	if err != nil {
+		srcDeploymentNeeded = true
+		dstDeploymentNeeded = true
+
```

Comment:
There is a case that Namespace and source deployment deployed successfully but destination deployment failed, how to handle this when user is trying to rerun the command?
Probably safe to add a label for the resources created by `antctl check` command and clean up stale resources every time when the command is re-triggered. 
cc @antoninbas @tnqn  for more thoughts

Reply:
I think we can add a Command flag for this purpose Which deletes The created namespace ?